### PR TITLE
[flashlight-cuda] update port to v0.3

### PR DESCRIPTION
**Pull request by jacobkahn**
Updates to Flashlight v0.3 (release https://github.com/flashlight/flashlight/releases/tag/v0.3).

Tested on Ubuntu and CentOS with gcc 7 (min compiler supported)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]

Same triplet support as before - no changes.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes

### CODE REVIEW

File: ports/flashlight-cuda/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "flashlight-cuda",
-  "version-string": "20210111",
+  "version-string": "0.3",
+  "port-version": 2,
```
Comment (JonLiu1993) [inline ports/flashlight-cuda/vcpkg.json:4]: @jacobkahn ,Thanks for your pr,please change "version-string" to "version",and delete "port-version"or set "port-version" to 0

File: versions/f-/flashlight-cuda.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "e9476e480b9f7562ee2fd30b7428a1e9c17f1881",
+      "version-string": "0.3",
```
Comment (JonLiu1993) [inline versions/f-/flashlight-cuda.json:5]: please change "version-string" to "version",and delete "port-version"or set "port-version" to 0

**Comment by JonLiu1993** [comment]
After changing the pr, please run the command "vcpkg x-add-version flashlight-cuda --overwrite-version", and then submit again

**Review by JonLiu1993 — APPROVED** [review]

File: versions/f-/flashlight-cuda.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "bd616669b5eb7413794d727b9909309ae686cd04",
+      "version": "0.3",
+      "port-version": 0
+    },
+    {
+      "git-tree": "e9476e480b9f7562ee2fd30b7428a1e9c17f1881",
+      "version-string": "0.3",
+      "port-version": 2
+    },
```
Comment (ras0219-msft) [inline versions/f-/flashlight-cuda.json:12]:
```suggestion
```
Each PR must add only one version to the database

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR! This looks good after my suggestion is applied.

**Comment by jacobkahn** [comment]
@JonLiu1993 @ras0219-msft can this be merged?

**Comment by BillyONeal** [comment]
I pushed a merge to this because we just changed the CUDA version installed on the VMs which is likely to affect this.

**Comment by BillyONeal** [comment]
I cancelled OSX build and am merging this anyway because our OSX fleet is degraded to 2 machines right now and this passed before on osx.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**